pennylane/math/autograd_box.py

```diff
@@ -78,6 +80,15 @@ def numpy(self):
     def requires_grad(self):
         return self.data.requires_grad
 
+    @wrap_output
+    def scatter_element_add(self, index, value):
+        size = self.data.size
+        flat_index = np.ravel_multi_index(index, self.shape)
+        t = [0] * size
+        t[flat_index] = value
+        self.data = self.data + np.array(t).reshape(self.shape)
+        return self.data
```

Comment:
The convoluted approach here is because autograd does not support array assignment, nor does numpy provide a functional approach to scattering :(

If anyone has any insights on a better way of doing this (while preserving differentiabilty in autograd), would appreciate it!